### ISSUE TITLE
JPMS

### ISSUE THREAD

**Issue by xenoterracide** [spring-projects/spring-ai#700]
please add jpms support, at least automatic modules for now. However, this issue or another issue should be added for full module-info at sometime in the future. Referencing this upstream spring issue (specifically comment) which shows all of the dependencies that could be included via the code generator https://github.com/spring-projects/spring-framework/issues/18079#issuecomment-2087396680

**Comment by markpollack** [comment]
Is this something that you could contribute?

**Comment by xenoterracide** [comment]
Unfortunately no, I don't currently have a use case for spring-ai. I simply took the list of spring projects that had not implemented an automatic-module and notified. I consider such a module to be unusable.

**Comment by markpollack** [comment]
I think we will wait on this, in particular to see how SF/SB boot move towards this new module system.

**Comment by xenoterracide** [comment]
Spring Boot and Spring and literally every other spring module have an automatic module in their manifest

**Comment by blacelle** [comment]
SpringBoot-4.1.0-M4: `MANIFEST.MF` :
```
Manifest-Version: 1.0
Automatic-Module-Name: spring.boot
Build-Jdk-Spec: 17
Built-By: Spring
Implementation-Title: Spring Boot
Implementation-Version: 4.1.0-M4
```

Spring-ai-model-2.0.0-M4:
```
Manifest-Version: 1.0
Created-By: Maven JAR Plugin 3.3.0
Build-Jdk-Spec: 25
Implementation-Title: spring-ai-model
Implementation-Version: 2.0.0-M4
```

Adding an automatic module name would be a great way forward.

As this projects seems based on maven, it may be as simple as:
```
<plugin>
    <artifactId>maven-jar-plugin</artifactId>
    <version>2.4</version>
    <configuration>
      <archive>
        <manifestEntries>
          <Automatic-Module-Name>org.springgramework.ai.spring-ai-model</Automatic-Module-Name>
        </manifestEntries>
      </archive>
    </configuration>
</plugin>
```
